lib/protocol/events.js

```diff
@@ -275,18 +273,29 @@ function eventsProtocolFactory (
         );
     };
 
-    EventsProtocol.prototype.publishEvent = function (data) {
+    EventsProtocol.prototype.publishExternalEvent = function (data) {
         assert.object(data);
         assert.string(data.type);
-
+        assert.string(data.action);
+        assert.string(data.severity);
+        assert.string(data.typeId);
+        assert.ok(data.hasOwnProperty("payload"));
+        if (data.nodeId !== null){
+            assert.string(data.nodeId);}
+
+        data.version = '1.0';
+        data.createdAt = new Date();
+
+        var routingKey = data.type + '.' + data.typeId + '.' + data.action +
+                         '.' + data.severity + '.' + data.nodeId;
```

Comment:
In this case, if nodeId is null or undefined, routingKey will be somthing like a.b.c.d.null or a.b.c.d.undefined, is this expected?